pydomcfg/domzgr/zco.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python
+
+"""
+Class to generate NEMO v4.0 standard geopotential z-coordinates
+"""
+
+from typing import Optional, Tuple
+
+import numpy as np
+from xarray import Dataset
+
+from pydomcfg.utils import is_nemo_none
+
+from .zgr import Zgr
+
+
+class Zco(Zgr):
+    """
+    Class to generate geopotential z-coordinates dataset objects.
+
+    Method
+    ------
+    *) Model levels' depths depT/W are defined from analytical function.
+    *) Model vertical scale factors e3 (i.e., grid cell thickness) can
+       be computed as
+
+             1) analytical derivative of depth function
+                (ln_e3_dep=False); for backward compatibility with v3.6.
+             2) discrete derivative (central-difference) of levels' depth
+                (ln_e3_dep=True). The only possibility from v4.0.
+
+    References:
+       *) NEMO v4.0 domzgr/zgr_z subroutine
+       *) Marti, Madec & Delecluse, 1992, JGR, 97, No8, 12,763-12,766.
+    """
+
+    # --------------------------------------------------------------------------
+    def __call__(
+        self,
+        ppdzmin: float,
+        pphmax: float,
```

Comment:
Maybe extend FORTRAN code and add something like `pphmax: Optional[float] = None`, if `None` we default to the maximum depth in the Bathymetry DataArray?

Reply:
I would prefer to leave like this to force the user the specify at least this two params - so he is aware of what he is doing, at least a bit

Reply:
Makes sense!